extensions/aws/s3/S3Wrapper.cpp

```diff
@@ -297,4 +404,57 @@ FetchObjectResult S3Wrapper::fillFetchObjectResult(const GetObjectRequestParamet
   return result;
 }
 
+void S3Wrapper::addListMultipartUploadResults(const Aws::Vector<Aws::S3::Model::MultipartUpload>& uploads, std::optional<std::chrono::milliseconds> max_upload_age,
+    std::vector<MultipartUpload>& filtered_uploads) {
+  const auto now = Aws::Utils::DateTime::Now();
+  for (const auto& upload : uploads) {
+    if (max_upload_age && now - upload.GetInitiated() <= *max_upload_age) {
+      logger_->log_debug("Multipart upload with key '%s' and upload id '%s' did not meet the age limit", upload.GetKey(), upload.GetUploadId());
+      continue;
+    }
```

Comment:
No, we actually want to skip those listed uploads that have not reached the age limit yet, because we only want to list the uploads that have already passed the max upload age, to delete the old pending uploads.

Reply:
So `multipartUploadExistsInS3()`, `listMultipartUploads()` and `addListMultipartUploadResults()` only list/check uploads which have aged off?  In that case, the names of the functions, and also local variables like `result` and `filtered_upload` should reflect this, to make the logic clearer.  Also, it would help people reading the code if the log message was something like "... has not aged off yet" instead of the vaguer ".. did not meet the age limit".

Reply:
Updated the variable names and added some comments in 5624af2c3ed7c36ecb55193c739d6845bf26d877